src/main/java/org/springframework/metrics/instrument/prometheus/PrometheusMeterRegistry.java

```diff
@@ -132,12 +136,38 @@ public double get() {
             };
 
             gauge.setChild(child, labelValues);
-            return new PrometheusGauge(name, child);
+            return new PrometheusGauge(id, child);
         });
 
         return obj;
     }
 
+    @Override
+    public MeterRegistry register(Meter meter) {
+        Collector collector = new Collector() {
+            @Override
+            public List<MetricFamilySamples> collect() {
+                List<MetricFamilySamples.Sample> samples = stream(meter.measure().spliterator(), false)
+                        .map(m -> {
+                            List<String> tagKeys = new ArrayList<>(m.getTags().size());
+                            List<String> tagValues = new ArrayList<>(m.getTags().size());
+                            for (Tag tag : m.getTags()) {
+                                tagKeys.add(tag.getKey());
+                                tagValues.add(tag.getValue());
+                            }
+                            return new MetricFamilySamples.Sample(m.getName(), tagKeys, tagValues, m.getValue());
+                        })
+                        .collect(toList());
+
+                return Collections.singletonList(new MetricFamilySamples(meter.getName(), Type.UNTYPED, " ", samples));
+            }
+        };
+        registry.register(collector);
+        collectorMap.put(meter.getName(), collector);
```

Comment:
I thought about this a bit too. Question is: what would we do if there was such a name already in the list? If the answer is blow up, `registry.register` is going to do that for us. Otherwise we are left with some sort of log and squash approach.